src/solvers/flattening/boolbv_overflow.cpp

```diff
@@ -161,7 +161,13 @@ literalt boolbvt::convert_overflow(const exprt &expr)
     return
       prop.land(!neg_shift, prop.lselect(undef, prop.lor(bv0), overflow));
   }
-  else if(
+
+  return SUB::convert_rest(expr);
+}
+
+literalt boolbvt::convert_unary_overflow(const unary_overflow_exprt &expr)
+{
+  if(
```

Comment:
There really is just this single `unary_minus_overflow_exprt` that exists for this case, so I would suggest to make this `convert_unary_minus_overflow(const unary_minus_overflow_exprt &expr)`.

Reply:
`unary_minus_overflow_exprt` and `unary_overflow_exprt` are currently equivalent. This fact could be used to simplify the way they are currently used, or even to remove `unary_overflow_exprt` and just keep `unary_minus_overflow_exprt`. However I think it would make it more difficult to add other kinds of unary overflow expressions. Adding more checks isn't something I am planning to work on but does seem worth considering. The following example unary operations might be considered for overflow checks -

- The existing `abs_exprt`.
- Increment operators `++` / `--`.
- The trigonometric `tan` / "tangent" operation.
- Could a narrowing cast be considered to overflow?